plugins/faq/src/config-loader.php

```diff
@@ -32,9 +32,9 @@ function register_custom_configs( array $configurations ) {
 		? 'post-type'
 		: 'taxonomy';
 
-	$runtime_config = (array) require_once _get_plugin_directory() . '/config/' . $filename . '.php';
+	$runtime_config = (array) require _get_plugin_directory() . '/config/' . $filename . '.php';
```

Comment:
One benefit of running the integration test in a fully integrated fashion is: hidden subtle problems are revealed.

When switching to apply the filter, it calls all plugin callbacks again. These callbacks were called already when the plugin loaded up via `'init'`, i.e. _before_ the tests started. 

What happened? The FAQ plugin was returning an invalid array which caused a fatal error because it didn't have a `'post_type'` key. 

Why? Think about it. Look at the code.

Reply:
@hellofromtonya Wow T, this is subtle stuff. I not only looked at the code, I had to go to the PHP Manual to understand the difference between `require` and `require_once`. I'm going to describe what _I think_ happened to generate the fatal error, and how the change from `require_once` to `require` solves this problem. Tell me if I understand the issue correctly. 

When the `init` hook fires initially, the filter event: 

>`apply_filters( 'add_custom_post_type_runtime_config', array() );` 

fires as well. Callbacks registered to that filter are called. Those callbacks include `register_custom_configs( array $configurations )` used in _both_ the `faq` and `tours` plugins. [ By the way, the other 4 custom plugins in the Cornerstone project that leverage `central-hub` ( 'cornerstone-members', 'events', 'reviews', and 'recordings' ) are designed similarly to `tours` and `faq`. ]

The control structure `require_once` will load the config file in `faq` and `tours` one time. In production, that control structure would be sufficient to load the config into memory for each plugin when active. 

Since we're calling the filter event `'add_custom_post_type_runtime_config'` a second time for the purpose of integration testing, `require_once` used in another plugin ( e.g. `faq` ) that relies on the same filter event will result in a fatal error. The post-type config called in `faq` will not be called. 

The solution is to change `require_once` to `require` in the `faq` plugin's function `register_custom_configs`. That allows the control structure to run again when the filter event is called during plugin testing, and avoids throwing a fatal error. 

Is my understanding correct?

Reply:
You got it.